### PR TITLE
Change pagination to patternfly component

### LINKED ISSUE #5185: replace pagination component with PF Pagination Element
PF has their own paginationWidget now. To avoid restyling the dropdown to look like a simple arrow, we should just use the ready-made component. Current implementation is entirely self coded and contains deprecated elements.

### PR DESCRIPTION
**Pull request by matborowczyk**
# Description

* Short description here *

closes #5185 

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [x] Code is clear and sufficiently documented
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )

Screenshot 2023-11-13 at 15 31 00

### CODE REVIEW

File: src/UI/Components/PaginationWidget/Provider.tsx [hunk 1]
```diff
@@ -1,7 +1,8 @@
 import React from "react";
+import { Pagination as PaginationComponent } from "@patternfly/react-core";
```
Comment (LukasStordeur) [inline src/UI/Components/PaginationWidget/Provider.tsx:2]: Was the name reassignment really needed?

File: src/UI/Utils/useFeatures.tsx [hunk 2]
```diff
@@ -11,7 +11,7 @@ const extractFeatures = (fileContent) => {
 };
 
 const useFeatures = (): string[] => {
-  const [features, setFeatures] = useState([]);
+  const [features, setFeatures] = useState(["instanceComposer"]);
```
Comment (LukasStordeur) [inline src/UI/Utils/useFeatures.tsx:14]: Probably not meant to be comitted?

File: src/UI/Components/PaginationWidget/Provider.tsx (hunk 1 above)
Comment (matborowczyk) [reply src/UI/Components/PaginationWidget/Provider.tsx:2]: We have interface with the same name, so I decided that is better to change this component name than hunt for interface across the app :D

Comment (LukasStordeur) [reply src/UI/Components/PaginationWidget/Provider.tsx:2]: Fair enough :D

File: src/UI/Utils/useFeatures.tsx [hunk 3]
```diff
@@ -11,7 +11,7 @@ const extractFeatures = (fileContent) => {
 };
 
 const useFeatures = (): string[] => {
-  const [features, setFeatures] = useState([]);
+  const [features, setFeatures] = useState([""]);
```
Comment (LukasStordeur) [inline src/UI/Utils/useFeatures.tsx:14]: It was an empty array initially, maybe best to keep it that way.

Comment (matborowczyk) [reply src/UI/Utils/useFeatures.tsx]: my bad

**Review by LukasStordeur — APPROVED** [review]

**Pull request closed without merging**